File: java/leetcode/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> findLevels(TreeNode root, List<List<Integer>> result, int depth) {
        if (root == null) {
            return result;
        }
        if (result.size() <= depth) {
            List<Integer> newRow = new ArrayList();
            result.add(newRow);
        }
        result.get(depth).add(root.val);
        if (root.left != null) {
            result = findLevels(root.left, result, depth + 1);
        }
        if (root.right != null) {
            result = findLevels(root.right, result, depth + 1);
        }
        return result;
    }

    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList();
        int depth = 0;
        return findLevels(root, result, depth);
    }
}

File: python/leetcode/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        result = []
        depth = 0
        return self.findLevels(root, result, depth)
    
    def findLevels(self, root, result, depth):
        if root is None:
            return result
        if len(result) <= depth:
            result.append([])
        result[depth].append(root.val)
        if root.left is not None:
            result = self.findLevels(root.left, result, depth + 1)
        if root.right is not None:
            result = self.findLevels(root.right, result, depth + 1)
        return result
